src/main/kotlin/com/openlattice/rhizome/core/service/ContinuousRepeatingTaskService.kt

```diff
@@ -0,0 +1,152 @@
+package com.openlattice.rhizome.core.service
+
+/*
+ * Copyright (C) 2020. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+
+import com.google.common.util.concurrent.ListeningExecutorService
+import com.hazelcast.core.HazelcastInstance
+import com.kryptnostic.rhizome.pods.HazelcastPod
+import org.slf4j.Logger
+import java.time.Instant
+import java.util.concurrent.Semaphore
+import java.util.concurrent.TimeUnit
+
+internal const val DEFAULT_BATCH_TIMEOUT_MILLIS = 120_000L
+
+abstract class ContinuousRepeatingTaskService<T: Any>(
+        private val executor: ListeningExecutorService,
+        hazelcastInstance: HazelcastInstance,
+        private val logger: Logger,
+        queueName: String,
+        lockingMapName: String,
+        parallelism: Int,
+        private val fillChunkSize: Int = 0,
+        private val batchTimeout: Long = DEFAULT_BATCH_TIMEOUT_MILLIS
+) {
+    private val locks = hazelcastInstance.getMap<T, Long>(lockingMapName)
```

Comment:
I would rather pass a HazelcastMap instead of the string, so we use Davids new class, something like that:
`lockingMap: HazelcastMap<T, Long>`
`lockingMap.getMap(hazelcastInstance)`

Reply:
Yes, passing an IMap / IQueue makes more sense. I'd tried this before and goofed somewhere, leaving me thinking that I couldn't do it but it looks like I can!